### PR TITLE
Tooltips

### PR DESCRIPTION
**Pull request by pizzalemon**
Add tooltips to some buttons and data displays, should improve usability. Along with that, deleted Antenna Tracker because that thing isn't going to be implemented. Also a conversion switcher on the actions page for the UAV altitude.

### CODE REVIEW

File: client/src/pages/FlightData/tabs/Actions.js [hunk 1]
```diff
@@ -84,8 +87,13 @@ const Actions = () => {
...
 					<Row>
-						<Box label="Altitude" content={Aaltitude.toFixed(2) + " ft"} />
-						<Box label="Ground Speed" content={AgroundSpeed.toFixed(2) + " mph"} />
+						<Box
+							label="Altitude"
+							content={AaltitudeIsGlobal ? AaltitudeGlobal.toFixed(2) + " ft MSL" : Aaltitude.toFixed(2) + " ft AGL"}
+							onClick={() => { setAaltitudeIsGlobal(!AaltitudeIsGlobal) }}
+							title="The plane's altitude. MSL refers to above mean sea level. AGL is the height from the ground where the plane is."
+						/>
+						<Box label="Ground Speed" content={AgroundSpeed.toFixed(2) + " mph"} title="Speed over only the surface of the earth, not taking height into account." />
```
Comment (krishnans2006) [inline client/src/pages/FlightData/tabs/Actions.js]: Maybe mention that it's GPS speed? That wording is kinda vague...

File: client/src/pages/FlightData/tabs/Actions.js [hunk 2]
```diff
@@ -253,10 +261,10 @@ const Actions = () => {
 			</Column>
 			<Column style={{ marginBottom: "1rem" }}>
 				<Row height="2.5rem">
-					<Button warning={true} color={darkred} onClick={() => httppost("/ugv/sethome")}>Set home?</Button>
+					<Button warning={true} color={darkred} onClick={() => httppost("/ugv/sethome")} title="Set the plane home position.">Set home?</Button>
```
Comment (krishnans2006) [inline client/src/pages/FlightData/tabs/Actions.js]: UGV = plane (?)

File: client/src/pages/FlightData/tabs/Quick.js [hunk 3]
```diff
@@ -96,37 +96,40 @@ const Quick = () => {
 					</Row>
 					<Row>
 						<Box label="Altitude"
-							 content={AaltitudeIsGlobal ? AaltitudeGlobal.toFixed(2) + " ft MSL" : Aaltitude.toFixed(2) + " ft AGL"}
-							 onClick={() => {setAaltitudeIsGlobal(!AaltitudeIsGlobal)}}
-							style={{ cursor: "pointer" }} />
+							content={AaltitudeIsGlobal ? AaltitudeGlobal.toFixed(2) + " ft MSL" : Aaltitude.toFixed(2) + " ft AGL"}
+							onClick={() => {setAaltitudeIsGlobal(!AaltitudeIsGlobal)}}
+							style={{ cursor: "pointer" }}
+							title="The plane's altitude. MSL refers to above mean sea level. AGL is the height from the ground where the plane is." />
```
Comment (krishnans2006) [inline client/src/pages/FlightData/tabs/Quick.js]: More specifically the home location altitude

File: client/src/pages/FlightData/tabs/Quick.js [hunk 4]
```diff
@@ -96,37 +96,40 @@ const Quick = () => {
...
 				<Row style={{ gap: "1rem" }}>
 					<Row>
 						<Box label="Ground Speed"
-							 content={((AspeedIsInKnots ? 0.868976 : 1) * AgroundSpeed).toFixed(2) + (AspeedIsInKnots ? " knots" : " mph")}
-							 onClick={() => {setAspeedIsInKnots(!AspeedIsInKnots)}}
-							 style={{ cursor: "pointer" }} />
+							content={((AspeedIsInKnots ? 0.868976 : 1) * AgroundSpeed).toFixed(2) + (AspeedIsInKnots ? " knots" : " mph")}
+							onClick={() => {setAspeedIsInKnots(!AspeedIsInKnots)}}
+							style={{ cursor: "pointer" }}
+							title="Speed over only the surface of the earth, not taking height into account." />
```
Comment (krishnans2006) [inline client/src/pages/FlightData/tabs/Quick.js]: Again, mention that it's GPS speed

File: client/src/pages/FlightData/tabs/Quick.js [hunk 5]
```diff
@@ -96,37 +96,40 @@ const Quick = () => {
...
 						<Box label="Airspeed"
-							 content={((AspeedIsInKnots ? 0.868976 : 1) * Aairspeed).toFixed(2) + (AspeedIsInKnots ? " knots" : " mph")}
-							 onClick={() => {setAspeedIsInKnots(!AspeedIsInKnots)}}
-							 style={{ cursor: "pointer" }} />
+							content={((AspeedIsInKnots ? 0.868976 : 1) * Aairspeed).toFixed(2) + (AspeedIsInKnots ? " knots" : " mph")}
+							onClick={() => {setAspeedIsInKnots(!AspeedIsInKnots)}}
+							style={{ cursor: "pointer" }}
+							title="Speed of the plane overall, including change in altitude." />
```
Comment (krishnans2006) [inline client/src/pages/FlightData/tabs/Quick.js]: That's not what airspeed is - it's speed data from sensors on the plane (relative to wind speed)

File: client/src/pages/FlightData/tabs/Quick.js [hunk 6]
```diff
@@ -96,37 +96,40 @@ const Quick = () => {
...
+						<Box label="Waypoint #" content={"#" + (Awaypoint[0] + 1).toFixed(0)} title="The waypoint number the plane is traveling to." />
+						<Box label="Distance" content={Awaypoint[1].toFixed(2) + " ft"} title="The distance to the next waypoint." />
 					</Row>
 					<Row>
-						<Box label="GPS HDOP" content={Aconnection[0].toFixed(2)} />
-						<Box label="GPS VDOP" content={Aconnection[1].toFixed(2)} />
-						<Box label="Satellites" content={Aconnection[2].toFixed(0)} />
+						<Box label="GPS HDOP" content={Aconnection[0].toFixed(2)} title="Horizontal dilution of precision. The higher, the less accurate the GPS is." />
+						<Box label="GPS VDOP" content={Aconnection[1].toFixed(2)} title="Vertical dilution of precision. The higher, the less accurate the GPS is." />
+						<Box label="Satellites" content={Aconnection[2].toFixed(0)} title="The number of satellites the plane is using." />
```
Comment (krishnans2006) [inline client/src/pages/FlightData/tabs/Quick.js]: Maybe mention thresholds for proper flight?

4 satellites is the absolute minimum, with 6 being reasonable, 8+ being good, and 10+ being very accurate

File: client/src/pages/FlightData/tabs/Quick.js [hunk 7]
```diff
@@ -137,8 +140,8 @@ const Quick = () => {
 			<Column style={{ marginBottom: "1rem", gap: "0.5rem" }}>
 				<Row style={{ gap: "1rem" }}>
 					<Row>
-						<Box label="Ground Speed" content={GgroundSpeed.toFixed(2) + " mph"} />
-						<Box label="Yaw" content={Gyaw + "\u00B0"} />
+						<Box label="Ground Speed" content={GgroundSpeed.toFixed(2) + " mph"} title="Speed over only the surface of the earth" />
```
Comment (krishnans2006) [inline client/src/pages/FlightData/tabs/Quick.js]: Again, mention GPS speed

File: client/src/pages/FlightData/tabs/Quick.js [hunk 8]
```diff
@@ -151,18 +154,18 @@ const Quick = () => {
 					</Row>
 					<Row>
 						<Box label="Status" content={Gstatus} />
-						<Box label="Mode" content={Gmode} />
+						<Box label="Mode" content={Gmode} title="The mode of mission the UGV is in, including RTL, and Auto mode." />
```
Comment (krishnans2006) [inline client/src/pages/FlightData/tabs/Quick.js:157]:
```suggestion
						<Box label="Mode" content={Gmode} title="The mode of mission the UGV is in, including RTL, Hold, and Auto mode." />
```

File: client/src/pages/FlightData/tabs/Quick.js [hunk 9]
```diff
@@ -151,18 +154,18 @@ const Quick = () => {
...
+						<Box label="To Destination" content={Gdestination.toFixed(2) + " ft"} title="Distance to the UGV's next waypoint." />
 						<Box label="Battery" content={Gbattery.toFixed(2) + "V"} />
 					</Row>
 					<Row>
-						<Box label="GPS HDOP" content={Gconnection[0].toFixed(2)} />
-						<Box label="GPS VDOP" content={Gconnection[1].toFixed(2)} />
-						<Box label="Satellites" content={Gconnection[2].toFixed(0)} />
+						<Box label="GPS HDOP" content={Gconnection[0].toFixed(2)} title="Horizontal dilution of precision. The higher, the less accurate the GPS is." />
+						<Box label="GPS VDOP" content={Gconnection[1].toFixed(2)} title="Vertical dilution of precision. The higher, the less accurate the GPS is." />
+						<Box label="Satellites" content={Gconnection[2].toFixed(0)} title="The number of satellites the UGV is using." />
```
Comment (krishnans2006) [inline client/src/pages/FlightData/tabs/Quick.js]: Again, add extra info on satellite count

File: client/src/pages/Submissions/Tabs.js [hunk 10]
```diff
@@ -18,11 +18,13 @@ const ViewRow = ({ checkboxes, info, i, active: [active, setActive], images: [im
 						style={{ gridArea: "accept" }}
 						type="accept"
 						callback={() => { accept(i) }}
+						title="Submit the ODLC to Interop."
 					/>
 					<Checkbox
 						style={{ gridArea: "decline" }}
 						type="decline"
                         callback={() => { reject(i) }}
+						title="Reject the ODLC."
```
Comment (krishnans2006) [inline client/src/pages/Submissions/Tabs.js]:
```suggestion
						title="Move the ODLC to a separate list of rejected ODLCs."
```

File: client/src/pages/Submissions/toolbarForm.js [hunk 11]
```diff
@@ -178,15 +178,15 @@ export const Checkboxes = ({ accept, save, decline, vertical, disabled }) => (
 	<Row style={{ marginBottom: "1rem" }}>
 		{vertical ?
 				<Column width="4rem" gap="0.5rem">
-					<Checkbox type="accept" callback={accept} disabled={disabled} />
-					<Checkbox type="save" callback={save} disabled={disabled} />
-					<Checkbox type="decline" callback={decline} disabled={disabled} />
+					<Checkbox type="accept" callback={accept} disabled={disabled} title="Submit the ODLC to Interop." />
+					<Checkbox type="save" callback={save} disabled={disabled} title="Save changes to the ODLC, but don't submit." />
+					<Checkbox type="decline" callback={decline} disabled={disabled} title="Reject the ODLC." />
```
Comment (krishnans2006) [inline client/src/pages/Submissions/toolbarForm.js]:
```suggestion
					<Checkbox type="decline" callback={decline} disabled={disabled} title="Move the ODLC to a separate list of rejected ODLCs." />
```

File: client/src/pages/Submissions/toolbarForm.js [hunk 12]
```diff
@@ -178,15 +178,15 @@ export const Checkboxes = ({ accept, save, decline, vertical, disabled }) => (
...
 				</Column>
 			:
 				<Row height="3rem" gap="0.5rem">
-					<Checkbox type="accept" callback={accept} disabled={disabled} />
-					<Checkbox type="save" callback={save} disabled={disabled} />
-					<Checkbox type="decline" callback={decline} disabled={disabled} />
+					<Checkbox type="accept" callback={accept} disabled={disabled} title="Submit the ODLC to Interop." />
+					<Checkbox type="save" callback={save} disabled={disabled} title="Save changes to the ODLC, but don't submit." />
+					<Checkbox type="decline" callback={decline} disabled={disabled} title="Reject the ODLC." />
```
Comment (krishnans2006) [inline client/src/pages/Submissions/toolbarForm.js]:
```suggestion
					<Checkbox type="decline" callback={decline} disabled={disabled} title="Move the ODLC to a separate list of rejected ODLCs." />
```

**Review by krishnans2006 — CHANGES_REQUESTED** [review]
Mostly just tooltip wording additions, changes, and fixes. Otherwise LGTM :D

File: client/src/pages/Submissions/toolbarForm.js (hunk 12 above)
Comment (pizzalemon) [reply client/src/pages/Submissions/toolbarForm.js]: I feel that this is well understood.

**Review by krishnans2006 — APPROVED** [review]
👍

**Pull request merged**